tests/UnixServerTest.php

```diff
@@ -268,8 +268,8 @@ public function testListenOnBusyPortThrows()
      */
     public function testConnectWithOverlyLongAddress()
     {
-        // string > 104 characters
-        $path = "unix://" . sys_get_temp_dir() . DIRECTORY_SEPARATOR . str_repeat('_', 100) . '.sock';
+        // string > 104/108 characters
+        $path = "unix://" . sys_get_temp_dir() . DIRECTORY_SEPARATOR . str_repeat('_', 104) . '.sock';
```

Comment:
It should probably subtract the length of the `sys_get_temp_dir()` for the `str_repeat`.

Reply:
It just needs to be long enough to trigger the exception? I did not want to go into complex scenarios here.

Reply:
Yes, but it should trigger on exactly X byte and succeed otherwise, no? Probably should have a tests that it works with one less character.

Reply:
Thanks for the discussion so far! I wonder if we really need this check in the first place? It looks like `stream_socket_server()` rejects an invalid path anyway, so why not rely on this?

https://3v4l.org/Gff8D

This would allow us to remove any platform specific constants / assumptions. What do you think about this?

Reply:
Never mind, it looks like it actually still succeeds despite this warning and instead creates a truncated socket path. Sorry for the confusion, go ahead :-)

Reply:
right, your test is invalid, see https://3v4l.org/4OPt0